libs/domain/picking/src/components/customer-note/customer-note.component.ts

```diff
@@ -32,7 +31,9 @@ export class CustomerNoteComponent extends PickingListMixin(LitElement) {
         ),
         catchError((e) => {
           if (e.status === 409) {
-            this.pickingInProgress = true;
+            const modal = this.pickingInProgressModal
+              .value as PickingInProgressModalComponent;
```

Comment:
You can move the type to the `createRef()` so the type cast is not needed here:
```
createRef<PickingInProgressModalComponent>()
```
Also the ref may be undefined so you need to check that it exists before assigning anything to it.